Calculator/CalculatorBrain.swift

```diff
@@ -18,14 +26,103 @@ class CalculatorBrain {
     var operatorType: OperatorType?
     
     func addOperandDigit(_ digit: String) -> String {
-        
+        if operatorType == nil {
+            if digit == ".", operand1String.contains(".") {
+                return operand1String
+            }
+            operand1String.append(digit)
+            return operand1String
+        } else {
+            if digit == ".", operand2String.contains(".") {
+                return operand2String
+            }
+            operand2String.append(digit)
+            return operand2String
+        }
     }
     
     func setOperator(_ operatorString: String) {
-        
+        operatorType = OperatorType(rawValue: operatorString)
     }
     
     func calculateIfPossible() -> String? {
+        guard operand1String != "", operand2String != "" else { return nil }
+        guard operatorType != nil else { return nil }
+        
+        var result: String?
+        
+        if let operand1 = Double(operand1String),
+            let operand2 = Double(operand2String) {
+            switch operatorType! {
+            case .addition:
+                result = String(operand1 + operand2)
+            case .subtraction:
+                result = String(operand1 - operand2)
+            case .multiplication:
+                result = String(operand1 * operand2)
+            case .division:
+                guard operand2 != 0 else {
+                    operand1String = ""
+                    operand2String = ""
+                    operatorType = nil
+                    return "Error"
+                }
+                result = String(operand1 / operand2)
+            }
+        }
+        
+        if result != nil {
+            result = removeDecimalIfInteger(result!)
+            operand1String = result!
+            operand2String = ""
+            operatorType = nil
+        }
+        
+        return result
```

Comment:
You can return `nil` here. Each of your cases should return the strings that the function is looking for.